privacyidea/lib/token.py

```diff
@@ -413,7 +418,7 @@ def get_tokens(tokentype=None, realm=None, assigned=None, user=None,
 def get_tokens_paginate(tokentype=None, realm=None, assigned=None, user=None,
                 serial=None, active=None, resolver=None, rollout_state=None,
                 sortby=Token.serial, sortdir="asc", psize=15,
-                page=1, description=None, userid=None):
+                page=1, description=None, userid=None, filterRealm=None):
     """
     This function is used to retrieve a token list, that can be displayed in
     the Web UI. It supports pagination.
```

Comment:
Hm, from a library design level, I'm not *too* happy about ``get_tokens_paginate`` having parameters ``user`` which holds a realm, ``realm`` *and* ``filterRealm`` :-) 
But I'm not sure if we can make this nicer easily, so maybe we just have to document this properly.

Reply:
The problem is that the function calls would get far more complex!

The thing is that "realm" and "allowed_realms" do not simply sum up. If "realm" is not contained in "allowed_realms", than such token will not be returned!

If you want to simplify the library call, the code, that is calling the library will get much more complicated and very prone to errors!

Reply:
Yes. I mean, we could adjust ``realm`` such that it accepts a list of realms (and returns tokens which are in one of these realms), but this would complicate the calling locations.
So yeah, let's add a ``allowed_realms`` parameter and document it nicely.

Reply:
The parameter looks good to me now! Could you add it to the docstring? :)

Reply:
Will do

Reply:
done in 09065737